Description: DODS: fix crash on URL that are not DODS servers
Author: Even Rouault <even dot rouault at spatialys dot com>
Origin: https://trac.osgeo.org/gdal/changeset/36175
Bug: https://trac.osgeo.org/gdal/ticket/6718
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gdal/+bug/1640360

--- a/frmts/dods/dodsdataset2.cpp
+++ b/frmts/dods/dodsdataset2.cpp
@@ -496,6 +496,8 @@ char **DODSDataset::CollectBandsFromDDSV
 /* -------------------------------------------------------------------- */
     BaseType *poVar = get_variable( GetDDS(), oVarName );
 
+    if( poVar == NULL )
+        return papszResultList;
     if( poVar->type() == dods_array_c )
     {
         poGrid = NULL;
@@ -1024,7 +1026,11 @@ DODSDataset::Open(GDALOpenInfo *poOpenIn
 /*      Did we get any target variables?                                */
 /* -------------------------------------------------------------------- */
         if( CSLCount(papszVarConstraintList) == 0 )
-            throw Error( "No apparent raster grids or arrays found in DDS.");
+        {
+            CPLDebug( "DODS", "No apparent raster grids or arrays found in DDS.");
+            delete poDS;
+            return NULL;
+        }
 
 /* -------------------------------------------------------------------- */
 /*      For now we support only a single band.                          */
